manifests/config.pp

```diff
@@ -55,4 +55,11 @@
     content => $server_config_real,
   }
 
+  if $::osfamily == 'RedHat' and ($version == 'present' or versioncmp($version, '5.9') < 0) {
+    file { '/usr/share/activemq/activemq-data':
+      ensure => link,
+      target => '/usr/share/activemq/data', 
```

Comment:
Same problem here with ActiveMQ 5.9.1 on EL7

activemq-5.9.1-2.el7.noarch

2015-08-20 20:19:13,627 [main           ] INFO  SharedFileLocker               - Database activemq-data/puppetserver4.hacklab/KahaDB/lock is locked... waiting 10 seconds for the database to be unlocked. Reason: java.io.IOException: Failed to create directory 'activemq-data/puppetserver4.hacklab/KahaDB'

Thanks for the patch jlambert121